src/vma/iomux/epfd_info.cpp

```diff
@@ -324,18 +324,28 @@ int epfd_info::add_fd(int fd, epoll_event *event)
 			errno = ENOMEM;
 			return -1;
 		}
-		m_p_offloaded_fds[m_n_offloaded_fds] = fd;
-		++m_n_offloaded_fds;
-		m_fd_info[fd].offloaded_index = m_n_offloaded_fds;
 
 		//NOTE: when supporting epoll on epfd, need to add epfd ring list
 		//NOTE: when having rings in pipes, need to overload add_epoll_context
 		unlock();
 		m_ring_map_lock.lock();
-		temp_sock_fd_api->add_epoll_context(this);
+		ret = temp_sock_fd_api->add_epoll_context(this);
 		m_ring_map_lock.unlock();
 		lock();
 
+		if (ret < 0) {
+			if (errno == EEXIST) {
+				__log_dbg("epoll_ctl: fd=%d is already registered with this epoll instance (%d)", fd, this->get_epoll_fd());
+			} else { // errno == ENOMEM
```

Comment:
need to change to explicit `if (errno == ENOMEM)`
... tomorrow we might add additional errno return values

and then you can add final `else { log("unknown errno=%d") }`